packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts

```diff
@@ -76,32 +73,28 @@ export class AuthenticatorComponent
       version: VERSION,
     });
 
-    const { authService, initializeMachine } = this.authenticator;
-
-    this.unsubscribeHub = listenToAuthHub(
-      authService,
-      async (data, service) => {
-        await defaultAuthHubHandler(data, service);
-        /**
-         * Hub events aren't properly caught by Angular, because they are
-         * synchronous events. Angular tracks async network events and
-         * html events, but not synchronous events like hub.
-         *
-         * On any notable hub events, we run change detection manually.
-         */
-        this.changeDetector.detectChanges();
-
-        /**
-         * Hub events that we handle can lead to multiple state changes:
-         * e.g. `authenticated` -> `signOut` -> initialState.
-         *
-         * We want to ensure change detection runs all the way, until
-         * we reach back to the initial state. Setting the below flag
-         * to true to until we reach initial state.
-         */
-        this.isHandlingHubEvent = true;
-      }
-    );
+    const { initializeMachine } = this.authenticator;
+
+    this.authenticator.hubSubject.subscribe(() => {
```

Comment:
Instead of `Authenticator` doing the hub listening, `AuthenticatorService` will do it. Then `Authenticator` here just `subscribes` to the service and runs UI specific logic below when notable hub event happens.